// This file is automatically generated by sila2codegen.proto_generator0.1.9
// :generation date: 2019-07-16 13:47:42.688931

// ---- PLEASE DO NOT MODIFY MANUALLY !! ---

syntax = "proto3";

import "SiLAFramework.proto";

package sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1;

service SyringeConfigurationController { 
	// Set syringe parameters. If you change the syringe in one device, you need to setup the new syringe parameters to get proper conversion of flow rate und volume units.
	rpc SetSyringeParameters(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.SetSyringeParameters_Parameters) returns (sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.SetSyringeParameters_Responses);

	// remark: Property Parameters are always Empty
	rpc Subscribe_InnerDiameter(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_InnerDiameter_Parameters) returns (stream sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_InnerDiameter_Responses);

	// remark: Property Parameters are always Empty
	rpc Subscribe_MaxPistonStroke(sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_MaxPistonStroke_Parameters) returns (stream sila2.de.cetoni.pumps.syringepumps.syringeconfigurationcontroller.v1.Subscribe_MaxPistonStroke_Responses);

}

// ------ Data Type definitions -------

// ---- Command Parameter and Response Data Type definitions ----

// Parameter Message: basic float/double type
message SetSyringeParameters_Parameters {
	sila2.org.silastandard.Real InnerDiameter = 1;
	sila2.org.silastandard.Real MaxPistonStroke = 2;
}

// Response Message: void / empty type
message SetSyringeParameters_Responses {
}

// ---- Property Parameter and Response Data Type definitions ----

// Inner diameter of the syringe tube in millimetres.
message Subscribe_InnerDiameter_Parameters { }

// InnerDiameter_Response:
//    :param request: gRPC request
//            :param context: gRPC context
//            :param response.InnerDiameter: Inner diameter of the syringe tube in millimetres.
//
message Subscribe_InnerDiameter_Responses {
	 sila2.org.silastandard.Real InnerDiameter = 1;
}

// The maximum piston stroke defines the maximum position the piston can be moved to before it slips out of the syringe tube. The maximum piston stroke limits the maximum travel range of the syringe pump pusher.
message Subscribe_MaxPistonStroke_Parameters { }

// MaxPistonStroke_Response:
//    :param request: gRPC request
//            :param context: gRPC context
//            :param response.MaxPistonStroke: The maximum piston stroke defines the maximum position the piston can be moved to before it slips out of the syringe tube. The maximum piston stroke limits the maximum travel range of the syringe pump pusher.
//
message Subscribe_MaxPistonStroke_Responses {
	 sila2.org.silastandard.Real MaxPistonStroke = 1;
}

